video/out/w32_common.c

```diff
@@ -1328,6 +1467,23 @@ static void thread_disable_ime(void)
     FreeLibrary(imm32);
 }
 
+static void dynamic_api_load(struct vo_w32_state *w32)
+{
+    if (!w32->api.shcore_dll)
+        w32->api.shcore_dll = LoadLibraryW(L"shcore.dll");
+    w32->api.pGetDpiForMonitor = !w32->api.shcore_dll ? NULL :
+                (void *)GetProcAddress(w32->api.shcore_dll, "GetDpiForMonitor");
+}
+
+static void dynamic_api_unload(struct vo_w32_state *w32)
+{
+    w32->api.pGetDpiForMonitor = NULL;
+    if (w32->api.shcore_dll) {
+        FreeLibrary(w32->api.shcore_dll);
+        w32->api.shcore_dll = NULL;
+    }
+}
```

Comment:
Just a minor comment: you don't need the unload part. shcore.dll can stay loaded forever. A LoadLibrary calls will merely increase the reference count each time (and the reference count is checked for overflow internally, so that's no issue). Also dynamic_api_load doesn't need to check whether it's already loaded, since this function is called from the init function.